homeassistant/components/light/qwikswitch.py

```diff
@@ -4,18 +4,34 @@
 For more details about this platform, please refer to the documentation at
 https://home-assistant.io/components/light.qwikswitch/
 """
-import logging
+from homeassistant.components.qwikswitch import (
+    QSToggleEntity, DOMAIN as QWIKSWITCH)
+from homeassistant.components.light import SUPPORT_BRIGHTNESS, Light
 
-DEPENDENCIES = ['qwikswitch']
+DEPENDENCIES = [QWIKSWITCH]
 
 
-# pylint: disable=unused-argument
-def setup_platform(hass, config, add_devices, discovery_info=None):
+async def async_setup_platform(hass, _, add_devices, discovery_info=None):
     """Add lights from the main Qwikswitch component."""
-    if discovery_info is None:
```

Comment:
This should be kept, I think. You don't want to continue unless setup is via discovery, right? Keep it and return if true.

Reply:
Can do, but should we not raise some exception if a user incorrectly configures one of these platforms? Or log the error then? I dont like failing silently.

Reply:
I think documentation is enough in this case. It's our standard procedure at the moment.